src/grammar-registry.js

```diff
@@ -601,9 +617,19 @@ module.exports = class GrammarRegistry {
 
   // Extended: Get all the grammars in this registry.
   //
+  // * `options` (optional) {Object}
+  //   * `textMateOnly` (optional) {Boolean} Set to ignore
+  //     [Tree-sitter](https://github.blog/2018-10-31-atoms-new-parsing-system/) grammars
+  //
   // Returns a non-empty {Array} of {Grammar} instances.
-  getGrammars() {
-    return this.textmateRegistry.getGrammars();
+  getGrammars(params) {
```

Comment:
I think that if we want to return Tree Sitter grammars by default, which I agree would be the cleaner choice long term, then we need `TreeSitterGrammar` to conform to [the documented public API](https://atom.io/docs/api/v1.38.2/Grammar) of the `Grammar` objects we claim we are returning from this method. I think the `onDidUpdate` method could be a no-op, but we'd want a compatible implementation for `tokenizeLine` and `tokenizeLines`. If we return objects from this method that don't conform to that interface, we'll basically be invalidating our documented API. So I see two choices, both acceptable:

* Change this method to *not* return Tree Sitter grammars by default
* Backfill support for the public `Grammar` API on `TreeSitterGrammar` (this would actually be my preference)

Reply:
I've backfilled the public methods. `onDidUpdate` does nothing, and the `tokenizeLine`(s) functions return a generic result, applying only the root scope to each line.